src/video-player/components/runtime.tsx

```diff
@@ -67,14 +72,21 @@ export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, set
           authoredState: true,
           aspectRatio
         });
-        if (authoredState.captionUrl) {
+        if (authoredState.captionUrl && player.textTracks().length === 0) {
           player.addRemoteTextTrack({
             kind: "captions",
             language: "en",
             label: "English",
             src: authoredState.captionUrl,
-            "default": true
+            default: captionsOnByDefault
           }, false);
+          // Store a ref to the captions track s we can check visibility later. There is no easy toggle event for captions show/hide
+          captionsTrackRef.current = player.textTracks()[0];
+          if (captionsTrackRef.current) {
+            captionsTrackRef.current.addEventListener("cuechange", (cue) => {
+              log("cue change", { videoUrl: authoredState.videoUrl });
```

Comment:
Would it be worth including what the cue change event actually was (which cue is now visible/whatever)?